sys/dev/nmdm/nmdm.c

```diff
@@ -416,6 +452,29 @@ nmdm_outwakeup(struct tty *tp)
 	taskqueue_enqueue(taskqueue_swi, &np->np_task);
 }
 
+static int
+nmdm_unload(void *data)
+{
+	struct nmdmsoftc *ns;
+	sbintime_t start_time = sbinuptime();
+	sbintime_t timeout = SBT_1S * 5; // Try to force unload for 5 seconds
```

Comment:
I think this is entirely valid, but I'd really prefer to accomplish it without the timeout by adding some plumbing through the tty later to drain the deferred destroy_dev taskqueue. kern_conf.c doesn't seem to have quite the KPI that I think I'd like to see.

Ideally, I think we'd have some kind of `tty_drain_freed` -> `destroy_dev_drain_pending` -> `taskqueue_drain` that we can call to force the async bits to run, then we can just EBUSY it if we still have nmdm devices when we get back.